### ISSUE TITLE
Nonexistent xsd in src/test/resources/files/rulesets/ruleset-refs.xml

### ISSUE THREAD

**Issue by AntonEvers** [phpmd/phpmd#545]
https://github.com/phpmd/phpmd/pull/232 adds a ruleset with a references to http://files.addiks.de/xmlns/phpmd.xsd and http://www.addiks.net/xmlns/pmd. These are unavailable, so there is no way to validate the XML against the XSD.

Does anybody know what was the content of these links and can they be included in http://pmd.sourceforge.net/ruleset_xml_schema.xsd if they are crucial to phpmd?

cc: @manuelpichler, @addiks 
related issue: https://github.com/magento/magento2/issues/13400

**Comment by addiks** [comment]
Sorry for this mess. I completely forgot that this schema exists. I guess the internet never forgets one's old mistakes...

The domain "files.addiks.de" was part of an old server-structure that i replaced a few years ago with something much simpler. I have put this one domain back online. If you put the lookup "78.46.98.40 files.addiks.de" into the /etc/hosts file of the server that is executing the validation it should work again. This is however just a temporary workaround.

The **real** solution should be that this custom schema is not needed anymore. Like jpevers said the parts from the custom schema should have been included into the main schema at the time of the merge. As far as i see the main XSD schema is not part of this repository (at least i do not find it here) so i cannot create a pull-request for that. Whoever controls the schema at sourceforge should update the XSD file. **After** that has been done, the reference to the custom schema must be removed from the [failing test-case][1] and be replaced with the main one like in [the other tests][2]. I have created new versions of both [the test-ruleset][3] and [the new schema][4] as github gist's.

[1]: https://github.com/phpmd/phpmd/blob/409b27696a096e80badde04f1796ce6c225e70ee/src/test/resources/files/rulesets/ruleset-refs.xml#L6
[2]: https://github.com/phpmd/phpmd/blob/409b27696a096e80badde04f1796ce6c225e70ee/src/test/resources/files/rulesets/refset4.xml#L6
[3]: https://gist.github.com/addiks/93b3035a2152bbc1d8db1cd2ed2aeaf7
[4]: https://gist.github.com/addiks/b5f015e833634b5bccacfc0f4699d7d5

**Comment by addiks** [comment]
I have now also updated the zonefile for "files.addiks.de". So after DNS updates (which might take a few hours) the domain should be available regularly again. An hosts-entry would then be not needed anymore.

**Comment by ajpevers** [comment]
@addiks thank you very much!

**Comment by ajpevers** [comment]
@manuelpichler what are your thoughts about merging @addiks' xsd into the one on sourceforge?

```xml
<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns="http://www.addiks.net/xmlns/pmd" targetNamespace="http://www.addiks.net/xmlns/pmd" elementFormDefault="qualified">
<xs:import namespace="http://pmd.sf.net/ruleset/1.0.0" schemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd"/>
<xs:element name="ruleset">
    <xs:complexType>
        <xs:sequence>
            <xs:element ref="description" minOccurs="1" maxOccurs="1"/>
            <xs:element ref="exclude-pattern" minOccurs="0" maxOccurs="unbounded"/>
            <xs:element ref="include-pattern" minOccurs="0" maxOccurs="unbounded"/>
            <xs:element ref="rule" minOccurs="1" maxOccurs="unbounded"/>
            <xs:element ref="php-includepath" minOccurs="0" maxOccurs="unbounded"/>
        </xs:sequence>
        <xs:attribute name="name" type="xs:string" use="required"/>
        <xs:attribute name="language" type="xs:string" use="optional">
            <xs:annotation>
                <xs:documentation source="version">1.0.0+</xs:documentation>
                <xs:documentation source="description">
                    Language on which this ruleset applies; e.g. "java" (default), "jsp"
                </xs:documentation>
            </xs:annotation>
        </xs:attribute>
    </xs:complexType>
</xs:element>
<xs:element name="php-includepath" type="xs:string"/>
</xs:schema>
```

**Comment by AntonEvers** [comment]
@manuelpichler still curious to know your opinion on the above matter.

**Comment by ravage84** [comment]
> http://pmd.sf.net/

This is not PHP Mess Detector's but Java's Mess Detector's (the original) Website.

**Comment by erfanimani** [comment]
Just thought I would mention as it's related, XML autocomplete in my IDE doesn't work because of this, and I was confused whether I had set up my ruleset wrong.

![image](https://user-images.githubusercontent.com/930470/182293363-392f70e9-d880-4074-bbc5-d62d695df3cc.png)
